### PR TITLE
tests: add spread test for refreshing from an old snapd and core18

### PR DESCRIPTION
**Pull request by stolowski**
Create core18 image from old snapd/core18 snaps and refresh to the latest snaps - first to core18/snapd from the store, then to the snapd from current tree.

### CODE REVIEW

File: tests/nested/manual/snapd-refresh-from-old/task.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+environment:
+  NESTED_BUILD_SNAPD_FROM_CURRENT: false
+  NESTED_IMAGE_ID: snapd-refresh-from-old
+  SNAPD_SNAP_URL: https://storage.googleapis.com/spread-snapd-tests/snaps/snapd_2.45.2_5760.snap
+  CORE18_SNAP_URL: https://storage.googleapis.com/spread-snapd-tests/snaps/core18_20191126_1279.snap
+
+prepare: |
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+  nested_prepare_env
```
Comment (sergiocazzolato) [inline tests/nested/manual/snapd-refresh-from-old/task.yaml:16]: this is done in the spread.yaml for each test, not needed to do it here

File: tests/nested/manual/snapd-refresh-from-old/task.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      break
+    fi
+    sleep 1
+  done
+
+  nested_exec "snap changes" | MATCH ".* Done .* Refresh snaps.*\"snapd\""
+  nested_exec "snap changes" | MATCH ".* Done .* Refresh snaps.*\"core18\""
+
+  echo "Now refresh snapd with current tree"
+  nested_copy "$PWD/snapd-from-deb.snap"
```
Comment (sergiocazzolato) [inline tests/nested/manual/snapd-refresh-from-old/task.yaml:63]: $PWD/ not needed

**Review by sergiocazzolato — APPROVED** [review]
Thanks for the test. Minor comments inline.

**Review by mvo5 — APPROVED** [review]
Nice one!

File: tests/nested/manual/snapd-refresh-from-old/task.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  nested_exec "sudo snap wait system seed.loaded"
+  nested_exec "snap list" | MATCH "snapd.*5760"
+  nested_exec "snap list" | MATCH "core18.*1279"
+
+  INITIAL_BOOT_ID=$(nested_get_boot_id)
+
+  # refresh to latest snapd from store, this will drop from ssh.
```
Comment (mvo5) [inline tests/nested/manual/snapd-refresh-from-old/task.yaml:40]: Hm, thinking about this again I wonder if maybe we should test going to the snapd from the PR directly. The reason is that with the test as is we will do "old-snapd -> snapd-from-store -> PR-snapd" so we could miss issues that only happen when going directly  "old-snapd -> PR-snapd" and then we only learn about this once snapd is in the store. Does this make sense? WDYT?

Comment (stolowski) [reply tests/nested/manual/snapd-refresh-from-old/task.yaml:40]: Yeah, that crossed my mind too at some point. Maybe I'll make two variants in the test.

Comment (stolowski) [reply tests/nested/manual/snapd-refresh-from-old/task.yaml]: Done.

**Pull request merged**